src/frontend/src/optimizer/plan_node/batch_hash_join.rs

```diff
@@ -61,18 +61,15 @@ impl BatchHashJoin {
     fn derive_dist(
         left: &Distribution,
         right: &Distribution,
-        predicate: &EqJoinPredicate,
+        _predicate: &EqJoinPredicate,
```

Comment:
If this is not used, just remove it? Clippy fixes need manual review :)

Reply:
maybe we will add some assert with the predicate later.